crates/searchers/openapi.yml

```diff
@@ -0,0 +1,674 @@
+                that is specified is dependent on the kind of liquidity.
+              allOf:
+                - $ref: "#/components/schemas/Address"
+            gasEstimate:
+              description: |
+                A rough approximation of gas units required to use this
+                liquidity on-chain.
+              type: integer
+
+    Auction:
+      description: |
+        The abstract auction to be solved by the searcher.
+      type: object
+      required:
+        - tokens
+        - orders
+        - liquidity
+        - effectiveGasPrice
+      properties:
+        id:
+          description: |
+            An opaque identifier for the auction. Will be set to `null` for
+            requests that are not part of an auction (when quoting token prices
+            for example).
+          type: string
+        tokens:
+          description: |
+            A map of token addresses to token information.
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/TokenInfo"
+        orders:
+          description: |
+            The solvable orders included in the auction.
+          type: array
+          items:
+            $ref: "#/components/schemas/Order"
+        liquidity:
+          description: |
+            On-chain liquidity that can be used by the solution.
+          type: array
+          items:
+            $ref: "#/components/schemas/Liquidity"
+        effectiveGasPrice:
+          description: |
+            The current estimated gas price that will be paid when executing a
+            settlement. Additionally, this is the gas price that is multiplied
+            with a settlement's gas estimate for solution scoring.
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        deadline:
+          description: |
+            The deadline by which a solution to the auction is required.
+            Requests that go beyond this deadline are expected to be cancelled
+            by the caller.
+          allOf:
+            - $ref: "#/components/schemas/DateTime"
+
+    JitOrder:
+      description: |
+        A just-in-time liquidity order included in a settlement.
+      type: object
+      required:
+        - sellToken
+        - buyToken
+        - receiver
+        - sellAmount
+        - buyAmount
+        - validTo
+        - appData
+        - feeAmount
+        - kind
+        - partiallyFillable
+        - sellTokenBalance
+        - buyTokenBalance
+        - signingScheme
+        - signature
+      properties:
+        sellToken:
+          $ref: "#/components/schemas/Token"
+        buyToken:
+          $ref: "#/components/schemas/Token"
+        receiver:
+          $ref: "#/components/schemas/Address"
+        sellAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        buyAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        validTo:
+          type: integer
```

Comment:
This is the signed valid to for a JIT order. I am reluctant to change it in the DTO as it would be "useless" work on both sides (on the solver side, they would convert the timestamp value that they sign to a date time for the DTO, and on the driver side we would convert the date time to a timestamp for encoding in a settlement).

I really wanted to keep `JitOrder` a 1:1 with signed protocol orders (as they get encoded into a settlement as-is, and we don't care about understanding any of the values). I see it as a "data-blob" that the SC understands, and not something that we need the data to be nice.